RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,68 @@
+# Release Checklist
+
+This is a list of the things that need to happen during a release.
+
+## Build a Release
+
+### Prepare the Changelog
+
+1. Open the associated milestone. All issues and PRs should be closed. If
+   they are not you should reassign all open issues and PRs to future
+   milestones.
+1. Go through the commit history since the last release. Ensure that all PRs
+   that have landed are marked with the milestone. You can use this to
+   show all the PRs that are merged on or after YYY-MM-DD:
+   `https://github.com/issues?q=repo%3Acloudflare%2Fkv-asset-handler+merged%3A%3E%3DYYYY-MM-DD`
+1. Go through the closed PRs in the milestone.
+1. Add this release to the `CHANGELOG.md`. Use the structure of previous
+   entries. If you use VS Code, you can use [this snippet](https://gist.github.com/EverlastingBugstopper/04d1adb99506388ff9d7abd8d0a82bc3) to insert new changelog sections. If it is a release candidate, no official changelog is needed, but testing instructions will be added later in the process.
```

Comment:
this snippet isn't quite right for this repo because it contains links to wrangler and not kv-asset-handler, you should be able to fork the gist and change those links and should be good